### PR TITLE
Add docker make commands and CI

### PR DESCRIPTION
**Pull request by rooneyshuman**
- Adds make commands to build and push docker image.
- Adds GH actions to run tests and checks on push.
- Fixes linting errors.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -18,11 +15,21 @@ jobs:
...
 
-    - name: Build
-      run: go build -v ./...
-
     - name: Test
-      run: go test ./...
+      run: make test
 
     - name: Benchmark
       run: go test -v -run=XXX -benchmem -bench=. ./...
```
Comment (logyball) [inline .github/workflows/go.yml:22]: might as well make this one:

```suggestion
      run: make benchmark
```

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -1,14 +1,11 @@
 name: Go
 
 on:
-  push:
-    branches: [ main ]
-  pull_request:
-    branches: [ main ]
+  [push]
```
Comment (logyball) [inline .github/workflows/go.yml:4]: NIT - not a blocker, but we sure we want this on every push?  Maybe just on PRs?  Every push might eat through GH actions time, although I don't know what our limit is these days organizationally

**Review by logyball — APPROVED** [review]
LGTM, two small things that are not blockers

File: .github/workflows/go.yml (hunk 2 above)
Comment (rooneyshuman) [reply .github/workflows/go.yml:4]: I'm cool with that. I'll change it to prs to main & our hackathon branch

File: .github/workflows/go.yml (hunk 1 above)
Comment (rooneyshuman) [reply .github/workflows/go.yml:22]: oh duh

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -2,13 +2,13 @@ name: Go
 
 on:
   push:
-    branches: [ main ]
+    branches: [ main, dev/hackathon-dec-2023 ]
```
Comment (inkel) [inline .github/workflows/go.yml:5]: Let's remember to remove this one once the hackathon finishes.

**Review by inkel — APPROVED** [review]

**Pull request merged**